### PR TITLE
Explicit check for prefix characters in release to use in Linux version check

### LINKED ISSUE #1131: TigerBeetle crashes on startup on Fedora while trying to parse Fedora kernel release name
On my Fedora machine I get:

```
Running: ./tigerbeetle format --cluster=0 --replica=0 --replica-count=1 /home/phil/tb/tigerbeetle/zig-cache/tmp/PbRC13ZcO2qopxVz/0_0.tigerbeetle
info(io): creating "0_0.tigerbeetle"...                                                                                                   info(io): allocating 1.13494873046875GiB...
error: InvalidVersion
/home/phil/tb/tigerbeetle/zig/lib/std/SemanticVersion.zig:117:73: 0x30d6ec in parse (tigerbeetle)
            for (id) |c| if (!std.ascii.isAlphanumeric(c) and c != '-') return error.InvalidVersion;
                                                                        ^
/home/phil/tb/tigerbeetle/src/io/linux.zig:35:25: 0x310029 in init (tigerbeetle)
        const version = try std.SemanticVersion.parse(release);
                        ^
/home/phil/tb/tigerbeetle/src/tigerbeetle/main.zig:90:22: 0x3111d5 in init (tigerbeetle)
        command.io = try IO.init(128, 0);
                     ^
/home/phil/tb/tigerbeetle/src/tigerbeetle/main.zig:110:9: 0x317c3f in format (tigerbeetle)
        try command.init(allocator, path, true);
        ^
/home/phil/tb/tigerbeetle/src/tigerbeetle/main.zig:52:28: 0x364e01 in main (tigerbeetle)
        .format => |*args| try Command.format(allocator, .{
```

Here's my kernel name:

```
$ uname --kernel-release
6.3.8-100.fc37.x86_64
```

### PR DESCRIPTION
**Pull request by eatonphil**
Closes #1131

### CODE REVIEW

**Comment by matklad** [comment]
I _think_ @kprotty would be a better reviewer here --- what we do here with uname feels shady, and indeed something like feature detection might be better, but I don't how pedantically correct io_uring support detection works.

File: src/io/linux.zig [hunk 1]
```diff
@@ -31,7 +31,24 @@ pub const IO = struct {
...
+            // std.SemanticVersion requires there be no extra
+            // characters after the major/minor/patch numbers. But
+            // some distributions have extra characters, such as this
+            // Fedora one: 6.3.8-100.fc37.x86_64.
+            var last_valid_version_character_index: usize = 0;
+            for (uts.release) |c| {
+                if (c == '.' or (c >= '0' and c <= '9')) {
+                    last_valid_version_character_index += 1;
+                    continue;
+                }
```
Comment (kprotty) [inline src/io/linux.zig]: Believe this should handle the case of more than 3 dots (i.e. custom kernel) by ignoring what's after as `5.15.90.1-microsoft-standard-WSL2` fails `std.SemanticVersion.parse`.

Comment (eatonphil) [reply src/io/linux.zig]: Gotcha!

Comment (eatonphil) [reply src/io/linux.zig]: Moved this into stdx.zig so I could add some unit tests.

**Review by kprotty — APPROVED** [review]

**Pull request merged**